src/game_battlealgorithm.cpp

```diff
@@ -644,17 +644,18 @@ void Game_BattleAlgorithm::AlgorithmBase::GetResultMessages(std::vector<std::str
 		out.push_back(GetParameterChangeMessage(IsPositive(), GetAffectedAgility(), Data::terms.agility));
 	}
 
-	std::vector<RPG::State>::const_iterator it = conditions.begin();
+	for (auto state_id: conditions) {
+		auto* state = ReaderUtil::GetElement(Data::states, state_id);
+		assert(state);
```

Comment:
this assert should be okay (instead of Output::Warning) because other code already filtered the states